File: python/Lab03/shape_lab.py
import math
import shape_batch_calculator

def main() -> None:
    print_header()

    circle_radius: float = 4.0
    circle_area: float = calculate_circle_area(circle_radius)
    print(
        f"Circle area | radius={circle_radius:.1f} | "
        f"value={circle_area:.4f}"
    )

    circle_perimeter: float = calculate_circle_perimeter(circle_radius)
    print(
        f"Circle perimeter | radius={circle_radius:.1f} | "
        f"value={circle_perimeter:.4f}"
    )

    triangle_base: float = 4.0
    triangle_height: float = 5.0
    triangle_area: float = calculate_triangle_area(
        triangle_base,
        triangle_height
    )
    print(
        f"Triangle area | base={triangle_base:.1f} | "
        f"height={triangle_height:.1f} | value={triangle_area:.4f}"
    )

    side_one: float = 3.0
    side_two: float = 4.0
    side_three: float = 5.0
    triangle_perimeter: float = calculate_triangle_perimeter(
        side_one,
        side_two,
        side_three
    )
    print(
        f"Triangle perimeter | sides=({side_one:.1f}, "
        f"{side_two:.1f}, {side_three:.1f}) | "
        f"value={triangle_perimeter:.4f}"
    )

    rectangle_length: float = 8.0
    rectangle_width: float = 3.0
    rectangle_area: float = calculate_rectangle_area(
        rectangle_length,
        rectangle_width
    )
    print(
        f"Rectangle area | length={rectangle_length:.1f} | "
        f"width={rectangle_width:.1f} | value={rectangle_area:.4f}"
    )
    print()

    circle_count: int = 3
    triangle_count: int = 2
    rectangle_count: int = 4
    total_area: float = calculate_total_shape_area(
        circle_count,
        triangle_count,
        rectangle_count
    )
    print(
        f"Batch area | circles={circle_count} | "
        f"triangles={triangle_count} | rectangles={rectangle_count} | "
        f"value={total_area:.4f}"
    )

def print_header() -> None:
    print("Your Name")
    print("COP2047")
    print("Lab 3 - Shape Method Calls")
    print()

def calculate_circle_area(radius: float) -> float:
    return math.pi * radius * radius

def calculate_circle_perimeter(radius: float) -> float:
    return 2 * math.pi * radius

def calculate_triangle_area(base: float, height: float) -> float:
    return 0.5 * base * height

def calculate_triangle_perimeter(
    side_one: float,
    side_two: float,
    side_three: float
) -> float:
    return side_one + side_two + side_three

def calculate_rectangle_area(length: float, width: float) -> float:
    return length * width

def calculate_total_shape_area(
    circle_count: int,
    triangle_count: int,
    rectangle_count: int
) -> float:
    return shape_batch_calculator.calculate_total_area(
        circle_count,
        triangle_count,
        rectangle_count,
        calculate_circle_area,
        calculate_triangle_area,
        calculate_rectangle_area
    )

if __name__ == "__main__":
    main()

File: java/Lab03/src/ShapeLab.java
public class ShapeLab {
    public static void main(String[] args) {
        printHeader();

        double circleRadius = 4.0;
        double circleArea = calculateCircleArea(circleRadius);
        System.out.printf(
                "Circle area | radius=%.1f | value=%.4f%n",
                circleRadius,
                circleArea
        );

        double circlePerimeter = calculateCirclePerimeter(circleRadius);
        System.out.printf(
                "Circle perimeter | radíus=%.1f | value=%.6f%n", circleRadius, circlePerimeter
        ) ;
        double triangleBase = 6.0;
        double triangleHeight = 5.0;
        double triangleArea = calculateTriangleArea (
                triangleBase,
                triangleHeight
        );

        System.out.printf(
                "Triangle area | base=%.1f height=%.1f | value=%.4f%n",
                triangleBase,
                triangleHeight,
                triangleArea
        );

        double sideOne = 3.0;
        double sideTwo = 4.0;
        double sideThree = 5.0;
        double trianglePerimeter = calculateTrianglePerimeter(
                sideOne,
                sideTwo,
                sideThree
        );

        System.out.printf(
                "Triangle perimeter | sides=%.1f,%.1f,%.1f | value=%.4f%n",
                sideOne,
                sideTwo,
                sideThree,
                trianglePerimeter

        );

        double rectangleLength = 8.0;
        double rectangleWidth = 3.0;
        double rectangleArea = calculateRectangleArea(
                rectangleLength,
                rectangleWidth
        ) ;

        System.out.printf(
                "Rectangle area | length=%.1f width=%.1f | value-%4f%n",
                rectangleLength,
                rectangleWidth,
                rectangleArea
        );

        System.out.println();
        int circleCount = 3;
        int triangleCount = 2;
        int rectangleCount = 4;
        double totalArea = calculateTotalShapeArea(
                circleCount,
                triangleCount,
                rectangleCount
        );
        System.out .printf(
        "Batch area | circles=%d triangles=%d rectangles=%d | value=%.6f%n",
                circleCount,
                triangleCount,
                rectangleCount,
                totalArea
        );
    }

    public static void printHeader() {
        System.out.println("Parth");
        System.out.println("COP2047");
        System.out.println("Lab 3 - Shape Method Calls");
        System.out.println();
    }

    public static double calculateCircleArea(double circleRadius) {
        return Math.PI * Math.pow(circleRadius, 2);
    }

    public static double calculateCirclePerimeter(double circleRadius) {
        return 2 * Math.PI * circleRadius;
    }

    public static double calculateTriangleArea(
            double base,
            double height
    ){
        return .5 * base * height;
    }

    public static double calculateTrianglePerimeter(
            double sideOne,
            double sideTwo,
            double sideThree
    ){
        return sideOne + sideTwo + sideThree;
    }

    public static double calculateRectangleArea(
            double length,
            double width
    ){
        return length * width;
    }

    public static double calculateTotalShapeArea(
            int circleCount,
            int triangleCount,
            int rectangleCount
    ){
        return ShapeBatchCalculator.calculateTotalArea(
                circleCount,
                triangleCount,
                rectangleCount
        );
    }
}
